lib/index.js

```diff
@@ -405,13 +405,37 @@ FSTree.prototype.match = function(globs) {
 };
 
 FSTree.prototype.changes = function() {
-  return this._changes.filter(change => {
-    return filterMatches(change[1], this.cwd, this.files, this.include, this.exclude);
-  }).map(change => {
-    return [change[0], change[1].replace(`${this.cwd}`, ''), change[2]];
-  });
+  if(this.srcTree && this._changes.length === 0) {
+    this._ensureEntriesPopulated();
+    return getChangesFromEntries(this.entries);
+  } else {
+    return this._changes.filter(change => {
+      return filterMatches(change[1], this.cwd, this.files, this.include, this.exclude);
+    }).map(change => {
+      return [change[0], change[1].replace(`${this.cwd}`, ''), change[2]];
+    });
+  }
 };
 
+
+function getChangesFromEntries (entries) {
+  return entries.map( entry => {
+    let operation;
+    let path = entry.basePath + entry.relativePath;
+
+    if(fs.lstatSync(path).isDirectory()) {
+      operation = "mkdir";
+    } else if(fs.lstatSync(path).isFile()) {
+      operation = "create";
+    } else {
+      throw new Error(`ENOENT: no such file or directory, ${path}`);
+    }
```

Comment:
we do not want to use `fs` here;   use `isFile` and `isDirectory`, imported at the top of this file.